### ISSUE TITLE
[3.3.1] Adaptive layers causes gaps when slicing

### ISSUE THREAD

**Issue by waechtjn** [Ultimaker/Cura#3828]
**Application Version**
3.3.1

**Platform**
Windows 10.0.16299 Build 16299 (2018-05-10)

**Printer**
Crealty CR-10 
[Project file and STL model](https://github.com/Ultimaker/Cura/files/2014334/Files.zip)

**Steps to Reproduce**
- Add the model from the STL file in the attached ZIP archieve
- Rotate the model by 90° so that the round/dome like part faces upwards:
![Orientation](https://user-images.githubusercontent.com/6500737/40196555-2aaaf1fc-5a11-11e8-95e3-5cbb44cbd62c.png)
- Slice using 0.2 layer height and adaptive layers with maximum variation of 0.1

**Actual Results**
The sliced part contains a gap:
![Gap present](https://user-images.githubusercontent.com/6500737/40196662-7e187030-5a11-11e8-8659-1145f2068ca6.png)

**Expected results**
The sliced part should not contain gaps (picture is for a maximum variation 0.099):
![Gap not present](https://user-images.githubusercontent.com/6500737/40196722-aa52a0bc-5a11-11e8-8e00-c86df5014037.png)

**Additional Information**
There are no gaps when slicing without adaptive layers or when setting maximum variation to 0.099 (as in the picture above).

The (ugly) gaps are actually visible in the printed result as well.

**Comment by ChrisTerBeke** [comment]
There's some known issues with Adaptive Layers. We won't be working on this soon though.

**Comment by Ghostkeeper** [comment]
This is not a problem with adaptive layers, as far as I can see, but only shows up with adaptive layers because it is slicing at a different Z coordinate where this happens to occur.

The real problem is that the polygon is like 2.25 line widths thick in that place, so it manages to fit 2 lines there but not the third.

**Comment by waechtjn** [comment]
I agree: the problem is also present without using adaptive layer height.

You can reproduce it using the same model as above by setting layer height to 0.225. This time the gap is at the top (and, thus, less visible on the screenshot; however, you can see the infill):
![Screenshot of a gap in the sliced model without using adaptive layer height](https://user-images.githubusercontent.com/6500737/41732858-7e2755a0-7582-11e8-8ed2-f17aaedf7501.png)

**Comment by Ghostkeeper** [comment]
Looking at this open issue again, I suppose this sort of thing should get filled as tiny gaps, shouldn't it?

If there is a small gap of less than half a line width, I don't really see what a slicer could do to fill that short of performing a physics relaxation to spread out the existing lines better, which is infeasible for the allowed computation time.

Is Tiny Gaps enabled in this print?

**Comment by GregValiant** [comment]
I'm cleaning up these old reports.  I'll go ahead and close this as the Arachne Engine in current versions cures this problem.
